src/System.Text.Primitives/System/Text/Parsing/InvariantUtf16/InvariantUtf16_UInt32.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Text.Utf8;
+
+namespace System.Text
+{
+    public static partial class PrimitiveParser2
```

Comment:
I know I sent you a sample with PrimitiveParser2, but it's because I forgot to remove the "2". I think we should just add these methods as overloads to the existing class.